apiserver/facades/client/charms/client.go

```diff
@@ -521,7 +521,24 @@ func (a *API) resolveOneCharm(arg params.ResolveCharmWithChannel, mac *macaroon.
 		return result
 	}
 	result.URL = resultURL.String()
-	result.Origin = origin
+
+	// We have an issue where the charmhub API can return "all" for architecture
```

Comment:
We have an issue that can cause us issues? :stuck_out_tongue:

Reply:
Yes, lots of issues